TestExperimentServerIntegration.test_experiments_with_start_time fails consistently

### Information

- **Qiskit IBM Provider version**: main branch, d0167a3b21460d12d7f353aad054c341167a1d78
- **Python version**: 3.8
- **Operating system**: Ubuntu 20.04

### What is the current behavior?

Running the test in Travis CI (within IBM) and locally fails with:

```
$ LOG_LEVEL=DEBUG python -m unittest -v test.ibm.experiment.test_experiment_server_integration.TestExperimentServerIntegration.test_experiments_with_start_time
session._log_request_info:DEBUG:2021-09-27 10:56:44,021: Endpoint: /device_components. Method: GET. 
test_experiments_with_start_time (test.ibm.experiment.test_experiment_server_integration.TestExperimentServerIntegration)
Test retrieving all experiments for a specific type. ... session._log_request_info:DEBUG:2021-09-27 10:56:44,744: Endpoint: /experiments. Method: POST. 
session._log_request_info:DEBUG:2021-09-27 10:56:45,167: Endpoint: /experiments. Method: GET. 
session._log_request_info:DEBUG:2021-09-27 10:56:45,390: Endpoint: /experiments. Method: GET. 
session._log_request_info:DEBUG:2021-09-27 10:56:45,578: Endpoint: /experiments. Method: GET. 
session._log_request_info:DEBUG:2021-09-27 10:56:45,730: Endpoint: /experiments. Method: GET. 
session._log_request_info:DEBUG:2021-09-27 10:56:45,850: Endpoint: /experiments. Method: GET. 
session._log_request_info:DEBUG:2021-09-27 10:56:46,025: Endpoint: /experiments. Method: GET. 
session._log_request_info:DEBUG:2021-09-27 10:56:46,198: Endpoint: /experiments/773f6c29-d511-43dc-8ee3-6ee4569e9821. Method: DELETE. 

======================================================================
FAIL: test_experiments_with_start_time (test.ibm.experiment.test_experiment_server_integration.TestExperimentServerIntegration) (title='before start, None')
Test retrieving all experiments for a specific type.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/osboxes/ibmq/qiskit-ibm/test/ibm/experiment/test_experiment_server_integration.py", line 211, in test_experiments_with_start_time
    self.assertEqual(found, expected,
AssertionError: False != True : Experiment 773f6c29-d511-43dc-8ee3-6ee4569e9821 (not)found unexpectedly when filter usingstart_dt=2021-09-26 09:56:44.744127-05:00, end_dt=None. Found=False

----------------------------------------------------------------------
Ran 1 test in 7.237s

FAILED (failures=1)
```

I did some debugging with the local recreate and figured out that in our `staging` database there were several experiments created yesterday in the same time window that the test query is hitting and with the default limit of 10 it's returning _those_ experiments rather than the experiment created in the test.


### Steps to reproduce the problem

Run the test locally using staging credentials:

```
export QISKIT_IBM_USE_STAGING_CREDENTIALS=True
export QISKIT_IBM_STAGING_API_TOKEN=***
export QISKIT_IBM_STAGING_API_URL=https://auth-dev.quantum-computing.ibm.com/api
export QISKIT_IBM_STAGING_HGP=ibm-q-internal/labgroup/mriedem
export QISKIT_IBM_STAGING_DEVICE=backend1
```

### What is the expected behavior?

Test should pass.

### Suggested solutions

Adding `experiment_type='qiskit_test'` to the query that the test is making fixes the issue, or increasing the limit to something like 50.

--- a/test/ibm/experiment/test_experiment_server_integration.py
+++ b/test/ibm/experiment/test_experiment_server_integration.py
@@ -199,7 +199,8 @@ class TestExperimentServerIntegration(IBMTestCase):
         for start_dt, end_dt, expected, title in sub_tests:
             with self.subTest(title=title):
                 backend_experiments = self.provider.experiment.experiments(
-                    start_datetime_after=start_dt, start_datetime_before=end_dt)
+                    start_datetime_after=start_dt, start_datetime_before=end_dt,
+                    experiment_type='qiskit_test')
                 found = False
                 for exp in backend_experiments:
                     if start_dt:

